Answers.md

```diff
@@ -1,2 +1,59 @@
 # Your responses to the short answer questions should be laid out here using Mark Down.
 ### For help with markdown syntax [Go here](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)
+Describe some of the differences between .forEach & .map.
+----
+
+-  forEach dose not return anythig to the caller, instead it just loop over the array. 
+-  map returns a new array after some action on the array it was called to, it returns a new array excetly of same number of items inside the original array. 
+
+
+Name five different Types in JavaScript. A Type is something that can represent data. What is so special about Arrays?
+----
+
+- Boolean-->  true or false;
+- Null --> represents nothing or emptyness. 
+- Undefined --> if something is declared but yet to be instantiated(no value set yet);
+- Number--> any real number decimal, float, integer, binary, hex etc. 
+- String--> values within quote are non-mutable string data, eg.  'Imran', '123' etc.
+
+Arrays are objects with mutability, they are comma separated collection of data types, with heterogeneous data acceptance. They have special methods in Js, like length, push, pop, shift and unshift, slice, splice...
+
+
+one special thing I found about array that they are called ***by reference*** in contrast to primitives(numbers). Meaning if we pass an array to a function and modify it, the original array will be mutated. Or in other words, we can say when we pass them we actually pass the memory address where they are located, so if the array is changed or modified, the original array is being modified. 
```

Comment:
very important to understand! and understanding memory & pointers will be a huge help in C, so kudos to you!